ibm/service/appconfiguration/resource_ibm_app_config_segment.go

```diff
@@ -0,0 +1,346 @@
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return nil
+	}
+
+	appconfigClient, err := getAppConfigClient(meta, parts[0])
+	if err != nil {
+		return err
+	}
+
+	options := &appconfigurationv1.GetSegmentOptions{}
+	options.SetSegmentID(parts[1])
+
+	result, response, err := appconfigClient.GetSegment(options)
+	if err != nil {
+		return fmt.Errorf("[DEBUG] GetSegment failed %s\n%s", err, response)
+	}
+
+	d.Set("guid", parts[0])
+
+	if result.Name != nil {
+		if err = d.Set("name", result.Name); err != nil {
+			return fmt.Errorf("[ERROR] Error setting name: %s", err)
+		}
+	}
+	if result.SegmentID != nil {
+		if err = d.Set("segment_id", result.SegmentID); err != nil {
+			return fmt.Errorf("[ERROR] Error setting segment_id: %s", err)
+		}
+	}
+	if result.Description != nil {
+		if err = d.Set("description", result.Description); err != nil {
+			return fmt.Errorf("[ERROR] Error setting description: %s", err)
+		}
+	}
+	if result.Tags != nil {
+		if err = d.Set("tags", result.Tags); err != nil {
+			return fmt.Errorf("[ERROR] Error setting tags: %s", err)
+		}
+	}
+	if result.CreatedTime != nil {
+		if err = d.Set("created_time", result.CreatedTime.String()); err != nil {
+			return fmt.Errorf("[ERROR] Error setting createdTime: %s", err)
+		}
+	}
+	if result.UpdatedTime != nil {
+		if err = d.Set("updated_time", result.UpdatedTime.String()); err != nil {
+			return fmt.Errorf("[ERROR] Error setting updatedTime: %s", err)
+		}
+	}
+	if result.Href != nil {
+		if err = d.Set("href", result.Href); err != nil {
+			return fmt.Errorf("[ERROR] Error setting href: %s", err)
+		}
+	}
+	if result.Rules != nil {
+		segmentRules := []map[string]interface{}{}
+		for _, ruleItem := range result.Rules {
+			segmentRulesItemMap := resourceIbmAppConfigSegmentRuleToMap(ruleItem)
+			segmentRules = append(segmentRules, segmentRulesItemMap)
+		}
+		if err = d.Set("rules", segmentRules); err != nil {
+			return fmt.Errorf("[ERROR] Error setting segment_rules: %s", err)
+		}
+	}
+	if result.Features != nil {
+		err = d.Set("features", resourceIbmAppConfigSegmentFeatureToMap(result.Features))
+		if err != nil {
+			return fmt.Errorf("[ERROR] Error setting features %s", err)
+		}
+	}
+	if result.Properties != nil {
+		err = d.Set("properties", resourceIbmAppConfigSegmentPropertiesToMap(result.Properties))
+		if err != nil {
+			return fmt.Errorf("[ERROR] Error setting properties %s", err)
+		}
+	}
+	return nil
+}
+
+func resourceIbmIbmAppConfigSegmentUpdate(d *schema.ResourceData, meta interface{}) error {
+	parts, err := flex.IdParts(d.Id())
+	if err != nil {
+		return nil
+	}
+	appconfigClient, err := getAppConfigClient(meta, parts[0])
+	if err != nil {
+		return err
+	}
+	options := &appconfigurationv1.UpdateSegmentOptions{}
+
+	options.SetSegmentID(parts[1])
+
+	if ok := d.HasChanges("name", "description", "tags", "rules", "attribute_name", "operator", "values"); ok {
```

Comment:
Does haschange work for inner elements? 
haschange of rules include its child elememts as well I believe..

Reply:
Yes I have tested haschange is working for inner elements of rules